nncf/quantization/algorithms/smooth_quant/torch_backend.py

```diff
@@ -0,0 +1,141 @@
+            om.PTConvTranspose3dMetatype,
+        ]
+
+    @property
+    def matmul_metatypes(self) -> List[OperatorMetatype]:
+        return [om.PTMatMulMetatype, om.PTLinearMetatype, om.PTModuleLinearMetatype]
+
+    @property
+    def quantize_agnostic_metatypes(self) -> List[OperatorMetatype]:
+        return DEFAULT_PT_QUANT_TRAIT_TO_OP_DICT[QuantizationTrait.QUANTIZATION_AGNOSTIC]
+
+    @staticmethod
+    def target_point(target_node_name: str, port_id: int) -> PTTargetPoint:
+        return PTTargetPoint(TargetType.OPERATOR_PRE_HOOK, target_node_name, input_port_id=port_id)
+
+    @staticmethod
+    def is_node_with_weights(node: NNCFNode) -> bool:
+        return node.layer_attributes is not None
+
+    @staticmethod
+    def get_activations_port_id(node: NNCFNode, nncf_graph: NNCFGraph) -> int:
+        return 0
+
+    @staticmethod
+    def get_channel_agnostic_reduction_axes(channel_axis: int, shape: Tuple[int]) -> Tuple[int]:
+        return get_channel_agnostic_reduction_axes([channel_axis], shape)
+
+    @staticmethod
+    def get_abs_max_channel_collector(
+        num_samples: int, stats_reduction_axes: Tuple[int], inplace: bool, branch_key: str
+    ) -> TensorCollector:
+        collector = TensorCollector()
+        reducer = PTAbsMaxReducer(reduction_axes=stats_reduction_axes)
+        aggregator = MaxAggregator(tensor_processor=PTNNCFCollectorTensorProcessor, num_samples=num_samples)
+        collector.register_statistic_branch(branch_key, reducer, aggregator)
+        return collector
+
+    @staticmethod
+    def get_weight_value(node_with_weight: NNCFNode, model: NNCFNetwork) -> Tensor:
+        node_module = model.nncf.get_containing_module(node_with_weight.node_name)
+        if node_module.weight is None:
+            return None
+        return Tensor(node_module.weight.data)
+
+    @staticmethod
+    def get_weight_tensor_port_id(node: NNCFNode) -> int:
+        const_ids = node.layer_attributes.get_const_port_ids()
+        if len(const_ids) != 1:
+            raise RuntimeError(f"Found more than 1 port for {node.node_name} node")
+        return const_ids[0]
+
+    @staticmethod
+    def weight_update_command(node_with_weight: NNCFNode, weight_value: np.ndarray) -> OVWeightUpdateCommand:
+        return create_command_to_update_weight(node_with_weight, weight_value)
+
+    @staticmethod
+    def scale_insertion_command(
+        source_node: NNCFNode,
+        scale_value: np.ndarray,
+        source_output_port_id: int,
+        nodes: List[NNCFNode],
+        scale_node_name: str,
+    ) -> OVMultiplyInsertionCommand:
+        input_port_id = 0
+        return multiply_insertion_command(nodes, scale_value, scale_node_name, input_port_id)
+
+    @staticmethod
+    def get_activation_channel_axis(node: NNCFNode, port_id: int) -> int:
+        if node.metatype == om.PTModuleLinearMetatype:
+            return -1
+        # TODO: Add activation axis calculation when MatMul wiil be supported
```

Comment:
Torch backend has distinguish between a linear layer and MatMul. As for linear layer, we know where its weights are kept (`.weight` attribute), where is the activation axis and etc. But the MatMul as an torch operator between two tensors needs more information to be smoothed.
Majority of networks (including all transformer models from the conformance test) are using `torch.Linear` module because it just easier to do. As we discussed this problem verbally, we decided to split SQ implementation and MatMul operations implementation on two to make code review process simpler.